lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,177 @@ class LogitFormatter(Formatter):
+
+    def use_overline(self, use_overline):
+        r"""
+        Switch display mode with overline for labelling p>1/2.
+
+        Parameters
+        ----------
+        use_overline : bool (default: False)
+            if x > 1/2, with x = 1-v, indicate if x should be displayed as
+            $\overline{v}$. The default is to display $1-v$.
+        """
+        self._use_overline = use_overline
+
+    def set_minor_threshold(self, minor_threshold):
+        """
+        Set the threshold for labelling minors ticks
+
+        Parameters
+        ----------
+        minor_threshold : int
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+        """
+        self._minor_threshold = minor_threshold
+
+    def set_minor_number(self, minor_number):
+        """
+        Set the number of minor ticks to label when some minor ticks are
+        labelled.
+
+        Parameters
+        ----------
+        minor_number : int
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._minor_number = minor_number
+
+    def set_locs(self, locs):
+        self.locs = locs
+        self._labelled.clear()
+
+        if not self._minor:
+            return None
+        if all(
+            is_decade(x, rtol=1e-7)
+            or is_decade(1 - x, rtol=1e-7)
+            or (is_close_to_int(2 * x) and int(np.round(2 * x)) == 1)
+            for x in locs
+        ):
+            # minor ticks are subsample from ideal, so no label
+            return None
+        if len(locs) < self._minor_threshold:
+            if len(locs) < self._minor_number:
+                self._labelled.update(locs)
             else:
-                s = '$1-{:.5f}$'.format(1-x)
+                # we do not have a lot of minor ticks, so only few decades are
+                # displayed, then we choose some (spaced) minor ticks to label.
+                alocs = np.array(locs)
+                diff = np.diff(-np.log(1 / alocs - 1))
+                space_pessimistic = np.minimum(
+                    np.concatenate(((np.inf,), diff)),
+                    np.concatenate((diff, (np.inf,))),
+                )
+                space_sum = (
+                    np.concatenate(((0,), diff))
+                    + np.concatenate((diff, (0,)))
+                )
+                good_minor = sorted(
+                    range(len(alocs)),
+                    key=lambda i: (space_pessimistic[i], space_sum[i]),
+                )[-self._minor_number:]
+                self._labelled.update(locs[i] for i in good_minor)
+
+    def _format_small_value(self, x, locs):
+        exponent = math.floor(np.log10(x))
+        value = x * 10 ** (-exponent)
+        if len(locs) < 2:
+            precision = 0
+        else:
+            diff = np.sort(np.abs(locs - x))[1]
+            precision = math.floor(-np.log10(diff) + exponent + 0.9)
```

Comment:
same comment here for the 0.9 as for the 0.1 earlier: is that really 1-1/base, or should that whole thing just use math.ceil?

Reply:
`math.ceil` is what I wanted to do, exempt log10 have imprecision effects. I change using ceil when it is a real float and round when it is near a int.

Fixed.